sdk/core/azure-core/azure/core/rest/_aiohttp.py

```diff
@@ -27,23 +27,42 @@
 import asyncio
 from typing import AsyncIterator
 from multidict import CIMultiDict
-from . import HttpRequest, AsyncHttpResponse
+from . import HttpRequest
+from ._http_response_impl_async import AsyncHttpResponseImpl
+from ._helpers import HeadersType
 from ._helpers_py3 import iter_raw_helper, iter_bytes_helper
 from ..pipeline.transport._aiohttp import AioHttpStreamDownloadGenerator
 
 
-class RestAioHttpTransportResponse(AsyncHttpResponse):
+class RestAioHttpTransportResponse(AsyncHttpResponseImpl):
     def __init__(
         self,
         *,
         request: HttpRequest,
         internal_response,
     ):
         super().__init__(request=request, internal_response=internal_response)
-        self.status_code = internal_response.status
-        self.headers = CIMultiDict(internal_response.headers)  # type: ignore
-        self.reason = internal_response.reason
-        self.content_type = internal_response.headers.get('content-type')
+        self._headers = CIMultiDict(internal_response.headers)  # type: ignore
+
+    @property
+    def status_code(self) -> int:
+        """The status code of this response"""
+        return self._internal_response.status
+
+    @property
+    def headers(self) -> HeadersType:
```

Comment:
We need to make sure that `HeadersType` is `import`ed from the "right" location (e.g. where we expect users of the library to import it from).

Reply:
that's a good point, do you think we should just stick with `MutableMapping`? Or do we have to go the whole route of the different transport's case insensitive dicts? I'm thinking `MutableMapping`, but not sure of your vision

Reply:
We should not expose the transport specific (concrete) implementations. Neither we nor the caller should care about that. The contract is that this is a mutable case insensitive multi dict. I'm good with typing this as a MutableMapping (it is unlikely that anything other than tests would see the difference) unless you really think we should do a [NewType](https://docs.python.org/3/library/typing.html#newtype) to explicitly call out the different behavior.

Reply:
I like `MutableMapping`, the only other new think that we should particularly highlight is htat it's "CaseInsensitive", but I think saying that in the docstrings etc is ok, idt we should do a whole new type for it, especially because we're more encouraging users to use lower case to access